src/units/UnitMergeReview.tsx

```diff
@@ -0,0 +1,156 @@
+import { useHistory } from 'react-router';
+
+import { ActionButtonContainer } from 'components/common/ActionButtonContainer';
+import { ConflictResolver } from 'components/common/ConflictResolver';
+import {
+  ConflictResolutionStrategy,
+  UnitComparison,
+  UnitsImportWithValidation,
+} from 'generated/sdk';
+import useDataApiWithFeedback from 'utils/useDataApiWithFeedback';
+
+interface UnitMergeReviewProps {
+  data: UnitsImportWithValidation;
+  onBack?: () => void;
+}
+
+const hasUnresolvedConflicts = (questionComparisons: UnitComparison[]) =>
+  questionComparisons.some(
+    (comparison) =>
+      comparison.conflictResolutionStrategy ===
+      ConflictResolutionStrategy.UNRESOLVED
+  );
+
+export function UnitMergeReview(props: UnitMergeReviewProps) {
+  const { api } = useDataApiWithFeedback();
+  const history = useHistory();
+  const templateExport = props.data;
+  const { version, json, errors } = templateExport;
+  const exportDate = dateformat(templateExport.exportDate, 'dd-mmm-yyyy');
+
+  const [state, setState] = useState({ ...templateExport });
+
+  const onConflictResolved = useCallback(
+    (
+      comparison: UnitComparison,
+      resolutionStrategy: ConflictResolutionStrategy
+    ) => {
+      setState(
+        produce((draft) => {
+          const updateQuestion = draft.unitComparisons.find(
+            (curComparison) =>
+              comparison.newUnit.id === curComparison.newUnit.id
+          );
+          if (updateQuestion) {
+            updateQuestion.conflictResolutionStrategy = resolutionStrategy;
+          }
+        })
+      );
+    },
+    []
+  );
+
+  const handleImportClick = () =>
+    api('Units imported successfully')
+      .importUnits({
+        json,
+        conflictResolutions: state.unitComparisons.map(
+          ({ newUnit, conflictResolutionStrategy }) => {
+            return {
+              itemId: newUnit.id,
+              strategy: conflictResolutionStrategy,
+            };
+          }
+        ),
+      })
+      .then(() => history.push('/Units'));
+
+  return (
+    <>
+      <Card>
+        <CardContent>
+          <Typography variant="body2">Version: {version}</Typography>
+          <Typography variant="body2">Export date: {exportDate}</Typography>
+        </CardContent>
+      </Card>
+      {errors.length !== 0 && (
+        <Alert severity="warning">
+          {errors.map((error) => (
+            <Typography key={error}>{error}</Typography>
+          ))}
+        </Alert>
+      )}
+
+      {props.data.unitComparisons.map((comparison) => (
+        <ConflictResolver<UnitComparison>
+          key={comparison.newUnit.id}
+          comparison={comparison}
+          onConflictResolved={onConflictResolved}
+          getStatus={(comparison) => comparison.status}
+          getItemId={(comparison) => comparison.newUnit.id}
+          getItemTitle={(comparison) => comparison.newUnit.unit}
+          getDiffInfo={({ existingUnit, newUnit }) => {
```

Comment:
Here also this returning array would be better to keep in a function and just call it here. My point is to clean up the html(tsx) part as much as possible because it looks better.